beps/pub-sub.rst

```diff
@@ -0,0 +1,88 @@
+:BEP: XXX
+:Title: Publish/Subscribe Protocol
+:Version: $Revision$
+:Last-Modified: $Date$
+:Author:  Steven Siloti <ssiloti@gmail.com>
+:Status:  Draft
+:Type:    Standards Track
+:Content-Type: text/x-rst
+:Created: 27-Aug-2016
+:Post-History: 
+
+
+Abstract
+========
+
+This BEP defines a modification of the DHT protocol [#BEP-5]_ to provide a topic based publish/subscribe service rather than a key/value store. Each topic is associated with a mutable item [#BEP-44]_. Messages are published as new values for the mutable item. A separate overlay network is maintained for each topic.
+
+
+Rationale
+=========
+
+BEP 46 [#BEP-46]_ defines a standard format for publishing a torrent which can later be updated by its publisher using a mutable item. A downside of this method is that it relies on polling the DHT to receive updates. The DHT must be polled at a relatively low rate to avoid putting excessive load on the network. This limitation means there may be a significant delay between an update being published and a client receiving it. This BEP aims to provide a means of propagating updates with minimal delay by using push messages rather than polling.
+
+
+Protocol
+========
+
+The pub/sub protocol is based on the DHT protocol [#BEP-5]_ with the following modifications:
+
+The value of K is set to one.
+
+The routing table's home bucket and its sibling SHOULD have a capacity of at least 8 nodes each. When the home bucket is split the sibling of the previous home bucket (now an uncle of the new home bucket) SHOULD be resized to one node per the setting of K=1.
+
+Every query MUST contain a key ``c`` with a 20-byte string value. The value is used to identify which topic the message is intended for. The value MUST be set to the target hash of the mutable item associated with the topic.
+
+The ``get_peers`` and ``announce_peer`` queries are prohibited.
+
+Mutable ``get`` and ``put`` queries [#BEP-44]_ are only permitted for the mutable item associated with the topic.
+
+The response to ``get`` queries SHOULD NOT contain a ``nodes`` or ``nodes6`` [#BEP-32]_ key.
```

Comment:
My thinking was that there is no point in traversing the network with `get` because all nodes have the item's value. Now that I think about it again though, it may make sense to prohibit `find_node` and use `get` instead. This would have the benefit of informing the requester if the node has a newer version of the item which the requester doesn't have for whatever reason.